File: matlab/autd3/Square.m
%{
%File: Square.m
%Project: autd3
%Created Date: 10/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 10/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef Square < Modulation

    methods

        function obj = Square(varargin)
            obj = obj@Modulation();
            f = varargin{1};

            if nargin < 2
                low = 0.0;
            else
                low = varargin{2};
            end

            if nargin < 3
                high = 1.0;
            else
                high = varargin{3};
            end

            if nargin < 4
                duty = 0.5;
            else
                duty = varargin{4};
            end

            pp = libpointer('voidPtrPtr', obj.ptr);
            calllib('autd3capi', 'AUTDModulationSquare', pp, cast(f, 'int32'), low, high, duty);
        end

    end

end

File: julia/src/Square.jl
# File: Square.jl
# Project: src
# Created Date: 14/06/2022
# Author: Shun Suzuki
# -----
# Last Modified: 14/06/2022
# Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
# -----
# Copyright (c) 2022 Shun Suzuki. All rights reserved.
# 

mutable struct Square
    _m::Modulation
    _header_ptr
    get_sampling_frequency_division
    set_sampling_frequency_division
    get_sampling_frequency
    function Square(freq; low::Float64=0.0, high::Float64=1.0, duty::Float64=0.5)
        chandle = Ref(Ptr{Cvoid}(0))
        autd3capi.autd_modulation_square(chandle, Int32(freq), low, high, duty)
        m = Modulation(chandle[])
        s = new(m, chandle[])
        s.get_sampling_frequency_division = m.get_sampling_frequency_division
        s.set_sampling_frequency_division = m.set_sampling_frequency_division
        s.get_sampling_frequency = m.get_sampling_frequency
        s
    end
end
